tools/windows/install-help/cal/delfiles.cpp

```diff
@@ -122,3 +136,146 @@ BOOL DeleteFilesInDirectory(const wchar_t* dirname, const wchar_t* ext, bool dir
     
     return TRUE;
 }
+
+
+/**
+ * This function recursively deletes all files in a given tree that match a given
+ * extension.  It will only accept an absolute path.
+ *
+ * @param dirname  The root path to start the deletion
+ *
+ * @param ext   the filename and/or extension to delete.  Can be a fixed name or wildcard
+ *
+ * @param dirs  If true, will delete directories which match the ext parameter, if the
+ *              directory is empty. If false (the default), will delete files
+ */
+BOOL DeleteHomeDirectory(std::wstring &user, PSID userSID)
+{
+    // first, find the path to the home directories
+    bool ret = false;
+    wchar_t * homeDir = NULL;
+    DWORD homeDirSize = _MAX_PATH;
+    bool needsLarger = false;
+    HANDLE hFind = INVALID_HANDLE_VALUE;
+    WIN32_FIND_DATA findFileData;
+    DWORD err;
+    std::wstring search;
+    do
+    {
+        if (homeDir) {
+            delete[] homeDir;
+        }
+        homeDir = new wchar_t[homeDirSize];
+        if (!GetProfilesDirectory(homeDir, &homeDirSize))
```

Comment:
The loop _in theory_ should only ever happen once (it's initial size is _MAX_PATH_).  The loop is provided to catch some sort of corner case.  But in the 99% case this will result in fewer calls to `GetProfilesDirectory()`